builder/azure/arm/step_deploy_template.go

```diff
@@ -233,31 +251,68 @@ func (s *StepDeployTemplate) deleteDeploymentResources(ctx context.Context, depl
 		resourceName := *deploymentOperation.Properties.TargetResource.ResourceName
 		resourceType := *deploymentOperation.Properties.TargetResource.ResourceType
 
-		s.say(fmt.Sprintf(" -> %s : '%s'", resourceType, resourceName))
+		s.say(fmt.Sprintf("Adding to deletion queue -> %s : '%s'", resourceType, resourceName))
+		resources[resourceType] = resourceName
+
+		if err = deploymentOperations.Next(); err != nil {
+			return err
+		}
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(resources))
```

Comment:
soo funny story. without the synchronization, Azure was returning the deployed resources in an order that happened to Just Work(TM) for deletion, incidentally ordering dependencies nicely. 

Adding this basically replicates what Azure does internally, but it does make you more likely to see some transient deletion failures during cleanup...I hope that's an acceptable compromise? The alternative would be sorting by resource type to make a semi-DAG of dependencies, which feels like a bit bigger change

Reply:
![image](https://user-images.githubusercontent.com/6800857/111564026-48029c80-8756-11eb-80c2-a3e99392992d.png)

how it looks right now with retries on failures

Reply:
(Incidentally, that means this logic was always racy and potentially broken, but somehow the likelihood of triggering it in a broken way was fairly low when doing serial deletion)